docs/extensibility.md

```diff
@@ -29,531 +28,630 @@ pub trait DiscoveryHandler {
 }
 ```
 
-To create a new protocol type, a new struct and impl block is required.  To that end, create a new folder for our Nessie code: `agent/src/protocols/nessie` and add a reference this new module in `agent/src/protocols/mod.rs`:
+To create a new protocol type, a new struct and impl block is required.  To that end, create a new folder for the HTTP code: `agent/src/protocols/http` and add a reference this new module in `agent/src/protocols/mod.rs`:
 
 ```rust
 mod debug_echo;
-mod nessie; // <--- Our new Nessie module
+mod http; // <--- Our new http module
 mod onvif;
 ```
 
-Next, add a few files to our new nessie folder:
+Next, add a few files to the new http folder:
 
-`agent/src/protocols/nessie/discovery_handler.rs`:
+To provide an implementation for the HTTP protocol discovery, create `agent/src/protocols/http/discovery_handler.rs` and define **HTTPDiscoveryHandler** and its HttpDiscoveryHandler.discover implementation.  For the HTTP protocol, the discovery handler will perform an HTTP GET on the protocol's discovery service URL:

```

Comment:
Not necessary, but maybe add more clarification/ emphasize the importance of the discover function. I felt a little thrown into the implementation. Could be something like this: "This function is called periodically by the Agent and returns the list of discovered devices. It should have all the functionality desired for discovering devices via your protocol and filtering for only the desired set. In our case, we will require that a URL is passed via the Configuration as a discovery endpoint. Our implementation will ping the discovery service at that URL to see if there are any "sensors""